src/solver/adaptivity.jl

```diff
@@ -0,0 +1,111 @@
+    ltg::LTG
+    σ_s::T
+    σ_c::T
+    Δt_bounds::NTuple{2,T}
+end
+
+mutable struct ReactionTangentControllerCache{T <: Real, LTGCache <: OS.LieTrotterGodunovCache, uType} <: OS.AbstractOperatorSplittingCache
+    const ltg_cache::LTGCache #It has Arrays so it can be const?
+    u::uType
+    uprev::uType # True previous solution
+    Rₙ₊₁::T
+    Rₙ::T
+    function ReactionTangentControllerCache(ltg_cache::LTGCache, Rₙ₊₁::T, Rₙ::T) where {T, LTGCache <: OS.LieTrotterGodunovCache}
+        uType = typeof(ltg_cache.u)
+        return new{T, LTGCache, uType}(ltg_cache, ltg_cache.u, ltg_cache.uprev, Rₙ₊₁, Rₙ)
+    end
+end
+
+@inline DiffEqBase.get_tmp_cache(integrator::OS.OperatorSplittingIntegrator, alg::OS.AbstractOperatorSplittingAlgorithm, cache::ReactionTangentControllerCache) = DiffEqBase.get_tmp_cache(integrator, alg, cache.ltg_cache)
+
+@inline function OS.advance_solution_to!(subintegrators::Tuple, cache::ReactionTangentControllerCache, tnext) 
+    OS.advance_solution_to!(subintegrators, cache.ltg_cache, tnext)
+end
+
+@inline DiffEqBase.isadaptive(::ReactionTangentController) = true
+
+"""
+    get_reaction_tangent(integrator::OS.OperatorSplittingIntegrator)
+Returns the maximal reaction magnitude using the [`PointwiseODEFunction`](@ref) of an operator splitting integrator that uses [`LieTrotterGodunov`](@ref) [Lie:1880:tti,Tro:1959:psg,God:1959:dmn](@cite).
+It is assumed that the problem containing the reaction tangent is a [`PointwiseODEFunction`](@ref).
+"""
+@inline function get_reaction_tangent(integrator::OS.OperatorSplittingIntegrator)
+    R, _ = _get_reaction_tangent(integrator.subintegrators)
+    return R
+end
+
+@inline @unroll function _get_reaction_tangent(subintegrators, n_reaction_tangents::Int = 0)
+    R = 0.0
+    @unroll for subintegrator in subintegrators
+        if subintegrator isa Tuple
+            R, n_reaction_tangents = _get_reaction_tangent(subintegrator, n_reaction_tangents)
+        elseif subintegrator.f isa PointwiseODEFunction
+            n_reaction_tangents += 1
+            φₘidx = transmembranepotential_index(subintegrator.f.ode)
+            R = max(R, maximum(@view subintegrator.cache.dumat[:, φₘidx]))
+        end
+    end
+    @assert n_reaction_tangents == 1 "No or multiple integrators using PointwiseODEFunction found"
+    return (R, n_reaction_tangents)
+end
+
+@inline function OS.stepsize_controller!(integrator::OS.OperatorSplittingIntegrator, alg::ReactionTangentController)
+    integrator.cache.Rₙ = integrator.cache.Rₙ₊₁
+    integrator.cache.Rₙ₊₁ = get_reaction_tangent(integrator)
+    return nothing
+end
+
+@inline function OS.step_accept_controller!(integrator::OS.OperatorSplittingIntegrator, alg::ReactionTangentController, q)
+    @unpack Rₙ₊₁, Rₙ = integrator.cache
+    @unpack σ_s, σ_c, Δt_bounds = alg
+    R = max(Rₙ, Rₙ₊₁)
+    if isinf(σ_s) && R == σ_c
+        # Handle it?
+        throw(ErrorException("Δt is undefined for R = σ_c where σ_s = ∞"))
```

Comment:
In the limit $\sigma_s -> \infty$ the sigmoidal function becomes a step function, i.e.
```suggestion
        integrator._dt = R > σ_c ? Δt_bounds[1] : Δt_bounds[2]
```

The analogue for IMEX methods would be the method found in

Whiteley JP. Physiology driven adaptivity for the numerical solution of the bidomain equations. Ann Biomed Eng. 2007; 35(9): 1510-1520. doi:[10.1007/s10439-007-9337-3](https://doi.org/10.1007/s10439-007-9337-3)